### PR TITLE
[Java] [Flight] [JDBC] Remove Deprecated Classes

### PR DESCRIPTION
**Pull request by iurysalino**
In this PR, I did changes in existents deprecated classes.

Remove the Deprecated classes `UrlSample.java`, `PropertiesSample.java`, and `FlightTestUtils.java`.

### CODE REVIEW

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java [hunk 1]
```diff
@@ -24,82 +24,60 @@
...
+    FLIGHT_SERVER_TEST_RULE = new FlightServerTestRule.Builder().host("localhost").randomPort()
+        .authentication(authentication).producer(PRODUCER).build();
+  }
+
+  private final String keyStorePath = this.getClass().getResource("/keys/keyStore.jks")
+      .getPath();
+  private final String noCertificateKeyStorePath =
+      this.getClass().getResource("/keys/noCertificate.jks")
+          .getPath();
+  private final String keyStorePass = "flight";
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:66]: Why setting this parameter there? By the name of the class, this just test a normal connection without tls

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java [hunk 2]
```diff
@@ -254,9 +237,10 @@ public void testUnencryptedConnectionShouldOpenSuccessfullyWithoutAuthentication
       throws Exception {
     final Properties properties = new Properties();
     properties.put(ArrowFlightConnectionProperty.HOST.camelName(), "localhost");
-    properties.put(ArrowFlightConnectionProperty.PORT.camelName(), server.getPort());
+    properties.put(ArrowFlightConnectionProperty.PORT.camelName(),
+        dataSource.getConfig().getPort());
     try (Connection connection = DriverManager
-        .getConnection(serverUrl, properties)) {
+        .getConnection("jdbc:arrow-flight://localhost:32010", properties)) {
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:243]: Is this test working? What I am reading by the test name is that you want to create a connection without using authentication. However, you are starting a server that needs authentication, so should this fail?

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java [hunk 3]
```diff
@@ -43,82 +40,59 @@
...
   private BufferAllocator allocator;
-  private FlightTestUtils flightTestUtils;
+  private ArrowFlightJdbcConnectionPoolDataSource dataSource;
 
   @Before
   public void setUp() throws Exception {
-    flightTestUtils = new FlightTestUtils("localhost", "flight1", "woho1",
-        "invalid", "wrong");
-
     allocator = new RootAllocator(Long.MAX_VALUE);
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java:87]: It seems that you are not using this allocator anymore. Please check it.

**Review by jcralmeida — CHANGES_REQUESTED** [review]

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java (hunk 1 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:66]: fixed

Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:66]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java (hunk 3 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java:87]: allocator is in use.

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java [hunk 4]
```diff
@@ -129,12 +103,12 @@ public void tearDown() throws Exception {
   @Test
   public void testGetEncryptedClientAuthenticated() throws Exception {
     final UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(
-        flightTestUtils.getUsername1(), flightTestUtils.getPassword1());
+        dataSource.getConfig().getUser(), dataSource.getConfig().getPassword());
```
[the remark this answers is not in the document]
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java:106]: fixed

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java [hunk 5]
```diff
@@ -43,82 +40,59 @@
...
-      identity = flightTestUtils.getUsername1();
-    } else {
-      throw CallStatus.UNAUTHENTICATED
-          .withDescription("Username or password is invalid.")
-          .toRuntimeException();
-    }
-    return () -> identity;
+    Collection<BufferAllocator> childAllocators = allocator.getChildAllocators();
+    AutoCloseables.close(childAllocators.toArray(new AutoCloseable[0]));
+    AutoCloseables.close(dataSource, allocator);
```
[the remark this answers is not in the document]
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java:95]: allocator is in use.

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java (hunk 2 above)
Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:200]: This test is correct. This test method is try to make connect unscripted and not necessary authentication.

Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:200]: I think is this...

Comment (iurysalino) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTest.java:200]: This test is correct. This test method is try to make connect unscripted and not necessary authentication.

I think is this...

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java [hunk 6]
```diff
@@ -166,6 +207,14 @@ private FlightServer getStartServer(Function<Location, FlightServer> newServerFr
     throw new IOException(exceptions.pop().getCause());
   }
 
+  public int getPort() {
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java:210]: Add java doc please

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java [hunk 7]
```diff
@@ -166,6 +207,14 @@ private FlightServer getStartServer(Function<Location, FlightServer> newServerFr
...
+    return config.getPort();
+  }
+
+  public String getHost() {
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java:214]: Add java doc please

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java [hunk 8]
```diff
@@ -300,4 +364,15 @@ public String getCookie() {
...
+
+    public final File cert;
+    public final File key;
+
+    public CertKeyPair(File cert, File key) {
+      this.cert = cert;
+      this.key = key;
+    }
+  }
+
```
Comment (jcralmeida) [inline java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java:377]: nit: remove extra empty line

File: java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java [hunk 9]
```diff
@@ -89,14 +98,35 @@ public static FlightServerTestRule createStandardTestRule(final FlightSqlProduce
...
+   * Create CertKeyPair object with the certificates and keys.
+   *
+   * @return A list with CertKeyPair.
+   */
+  public static List<CertKeyPair> exampleTlsCerts() throws URISyntaxException {
+    final Path root = getFlightTestDataRoot();
+    return Arrays.asList(
+        new CertKeyPair(root.resolve("cert0.pem").toFile(), root.resolve("cert0.pkcs1").toFile()),
+        new CertKeyPair(root.resolve("cert1.pem").toFile(), root.resolve("cert1.pkcs1").toFile()));
+  }
```
[the remark this answers is not in the document]
Comment (jcralmeida) [reply java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightServerTestRule.java:127]: Did you fix this?

**Review by jcralmeida — CHANGES_REQUESTED** [review]

**Pull request merged**